tesseract/integration-test/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+				match fetched_events {
+					Some(res_event) => {
+						post_request_handled_event = Some(res_event.clone());
+						break
+					},
+					None => continue,
+				}
+			},
+			Err(err) => {
+				panic!("Error in finalized block stream: {:?}", err)
+			},
+		}
+	}
+	log::info!("Chain B Event: {:?}", post_request_handled_event);
+	// The relayer should finish sending the request message to chain B
+
+	let bob_chain_b_new_balance = client_b
+		.storage()
+		.at_latest()
+		.await?
+		.fetch(&bob_key)
+		.await?
+		.ok_or("Failed to fetch")
+		.unwrap()
+		.data
+		.free;
+
+	// diving by 100000000000 for better assertion as the rem balance = initial - amount - fees
+	// in chain A
+	assert_eq!(
+		(bob_chain_a_initial_balance - amount) / 1000000000000,
+		bob_chain_a_new_balance / 1000000000000
+	);
+	// in chain B
+	assert_eq!(
+		(bob_chain_b_initial_balance + amount) / 1000000000000,
+		bob_chain_b_new_balance / 1000000000000
+	);
+
+	Ok(())
+}
+
+/// fetch a foreign storage item from a given key
+async fn get_request_works() -> Result<(), anyhow::Error> {
+	let _ = log_setup();
+	let (chain_a_sub_client, chain_b_sub_client) = create_clients().await?;
+
+	log::info!(" \n 🧊integration test for para: 2000 to para 2001: get request \n");
+
+	// =======================================================================
+	let (chain_a_client, chain_b_client) =
+		(Arc::new(chain_a_sub_client.clone()), Arc::new(chain_b_sub_client.clone()));
+
+	let (client_a, client_b) =
+		(chain_a_sub_client.clone().client, chain_b_sub_client.clone().client);
+
+	// Accounts & keys
+	let dave_signer = PairSigner::<Hyperbridge, _>::new(
+		Pair::from_string("//Dave", None).expect("Unable to create Dave account"),
+	);
+	// parachain info pallet fetching para id
+	let encoded_chain_b_id_storage_key =
+		"0x0d715f2646c8f85767b5d2764bb2782604a74d81251e398fd8a0a4d55023bb3f";
+
+	let latest_height_b =
+		chain_a_client.query_latest_height(chain_b_client.state_machine_id()).await? - 5;
+
+	let get_request = api::tx().ismp_demo().get_request(GetRequest {
+		para_id: 2001,
+		height: latest_height_b,
+		timeout: 0,
+		keys: vec![hex::decode(encoded_chain_b_id_storage_key.strip_prefix("0x").unwrap()).unwrap()],
+	});
+
+	let tx_result = client_a
+		.tx()
+		.sign_and_submit_then_watch_default(&get_request, &dave_signer)
+		.await?
+		.wait_for_finalized_success()
+		.await?
+		.all_events_in_block()
+		.clone();
+
+	let tx_block_hash = tx_result.block_hash();
+	let tx_block_height = client_a.blocks().at(tx_block_hash).await?.number() as u64;
+	let events = client_a.events().at(tx_block_hash).await?;
+	let event = events.find_last::<RequestEvent>()?.unwrap();
+	log::info!("Ismp Events: {:?} \n", event);
+
+	// ======================= handle the get request and resubmit to chain A (origin chain)
+	// =====================================
+
+	let value_returned_encoded =
+		relay_get_response_message(chain_a_sub_client, chain_b_sub_client, tx_block_height).await?;
```

Comment:
You already have the get request, why not just pass it on to `relay_get_response_message`